XCUITests/BaseTestCase.swift

```diff
@@ -10,20 +10,23 @@ class BaseTestCase: XCTestCase {
         super.setUp()
         continueAfterFailure = false
         let app = XCUIApplication()
-        restart(app)
+        app.terminate()
+        restart(app, args: [LaunchArguments.ClearProfile, LaunchArguments.SkipIntro, LaunchArguments.SkipWhatsNew])
```

Comment:
I am no expert on this, just curious, what if we add those parameters in the Test section of the scheme rather than killing the app and restarting it?would it save more time?
![screen shot 2017-10-03 at 12 21 06 am](https://user-images.githubusercontent.com/6930774/31109880-ca306750-a7d0-11e7-9991-fd9fec94cbb8.png)

Reply:
From fiddling with the [`app.launch()`][1] and [`app.terminate()`][2], it would appear that it's quite difficult to make an app last over more than one test without an effective re-install.

So, the `CLEAR_PROFILE` would appear effectively redundant.

We could make `CLEAR_PROFILE` more robust, but honestly, I'd take test integrity over test speed any day.

As to passing `launchArguments` as config data in the scheme: I'd rather this live with the code than with the launch configuration — developers are likely to run a single test in a different configuration. Either way, this will only make micro-seconds difference.

[1]: https://developer.apple.com/documentation/xctest/xcuiapplication/1500467-launch?changes=latest_major
[2]: https://developer.apple.com/documentation/xctest/xcuiapplication/1500637-terminate?changes=latest_major

Reply:
Sounds good